packages/devtools_app/lib/src/screens/memory/memory_snapshot_models.dart

```diff
@@ -943,9 +945,10 @@ FieldReference listToFieldEntries(
               );
 
               FieldReference valueEntry;
-              final HeapGraphClassLive valueClass = valueElement.theClass;
-              predefined = predefinedClasses[valueClass.fullQualifiedName];
-              if (predefined != null && predefined.isScalar) {
+              final HeapGraphClassLive valueClass =
+                  valueElement.theClass as HeapGraphClassLive;
+              predefined = predefinedClasses[valueClass.fullQualifiedName]!;
```

Comment:
maybe we should remove the null assert and go back to checking for null in the if below. This seems safer

Reply:
done